src/openrtb/handler.go

```diff
@@ -28,18 +28,121 @@ import (
 	"owid"
 	"swan"
 	"sync"
+	"github.com/bsm/openrtb"
+	"encoding/json"
+	"encoding/base64"
+	"log"
 )
 
 var empty swan.Empty // Used for empty responses
 
 const openRTBPath = "/demo/api/v1/bid" // The path for this handler
 
+const prebidRTBPath = "/demo/api/v1/prebid" // The path for prebid
+
+type ImpExt struct {
+	Offer owid.Node
+	Offername string
+}
+
 // Handler is responsible for a real time transaction for advertising.
 // The body of the request must contain a JSON array of Processor IDs which
 // contain the signature of the last entry in the list of Processors.
 func Handler(d *common.Domain, w http.ResponseWriter, r *http.Request) {
 
-	if r.URL.Path == openRTBPath && r.Method == "POST" {
+	if r.URL.Path == prebidRTBPath && r.Method == "POST" {
+		var req *openrtb.BidRequest
+		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+			log.Fatal(err)
+		}
+
+		if d.Config.Debug {
+			fmt.Println(d.Host)
+			fmt.Println(req)
+		}
+	
+		var bids = make([]openrtb.Bid, len(req.Impressions))
+		for i, e := range req.Impressions {
+			var raw ImpExt
+			if err := json.Unmarshal(e.Ext, &raw); err != nil {
+					panic(err)
+			}
+
```

Comment:
the bidder reads the offer (which should just be the original offer signed by the publisher) from imp.ext, and passes it to the existing handle transaction function.